components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorage.kt

```diff
@@ -48,8 +50,8 @@ abstract class PlacesStorage(
         RustPlacesConnection
     }
 
-    internal val writer: PlacesWriterConnection by lazy { places.writer() }
-    internal val reader: PlacesReaderConnection by lazy { places.reader() }
+    internal open val writer: PlacesWriterConnection by lazy { places.writer() }
+    internal open val reader: PlacesReaderConnection by lazy { places.reader() }
```

Comment:
@Mugurell One thing that I don't see anymore in this latest push is the separate reader for queries. Did you decide to switch back to just using the default one because the queries are serialized anyway?

Reply:
Thank you for raising this.
In the first attempt for this I was using a `newReader` to ensure `interrupt`ing the one used for history queries doesn't cancel other places queries. 
Was on the wrong assumption that all places queries would otherwise use the same reader which is cached in `Connection`. But this is not the case. [`PlacesHistoryStorage` is a `PlacesStorage`](https://github.com/mozilla-mobile/android-components/blob/1b490c833c9935a250b7b4c1e29eb7f272d3e75f/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt#L46) and [`PlacesBookmarksStorage` is separate a `PlacesStorage`](https://github.com/mozilla-mobile/android-components/blob/1b490c833c9935a250b7b4c1e29eb7f272d3e75f/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesBookmarksStorage.kt#L24) with [both creating their own places connection](https://github.com/mozilla-mobile/android-components/blob/70892aea03f48d9aa2ba11acc427df98329dab7e/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorage.kt#L46-L49) which so [caches the reader (and writer) at each `PlacesStorage` implementation level](https://github.com/mozilla-mobile/android-components/blob/b827bd3c9c90a315001be415380e8d748894017d/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/Connection.kt#L72-L88).